crates/primitives/starknet/src/execution/felt252_wrapper.rs

```diff
@@ -0,0 +1,373 @@
+        H256::from_slice(&buf)
+    }
+}
+
+/// FieldElement converter.
+impl From<FieldElement> for Felt252Wrapper {
+    fn from(ff: FieldElement) -> Self {
+        Self(ff)
+    }
+}
+
+/// FieldElement converter.
+impl From<Felt252Wrapper> for FieldElement {
+    fn from(ff: Felt252Wrapper) -> Self {
+        ff.0
+    }
+}
+
+/// Felt252 converter.
+impl From<Felt252> for Felt252Wrapper {
+    fn from(value: Felt252) -> Self {
+        Felt252Wrapper::try_from(&value.to_be_bytes()).unwrap()
+    }
+}
+
+/// Felt252 converter.
+impl From<Felt252Wrapper> for Felt252 {
+    fn from(felt: Felt252Wrapper) -> Self {
+        let buf: [u8; 32] = felt.into();
+        Felt252::from_bytes_be(&buf)
+    }
+}
+
+/// SCALE trait.
+impl Encode for Felt252Wrapper {
+    fn encode_to<T: Output + ?Sized>(&self, dest: &mut T) {
+        dest.write(&self.0.to_bytes_be());
+    }
+}
+
+/// SCALE trait.
+impl EncodeLike for Felt252Wrapper {}
+
+/// SCALE trait.
+impl MaxEncodedLen for Felt252Wrapper {
+    fn max_encoded_len() -> usize {
+        32
+    }
+}
+
+/// SCALE trait.
+impl Decode for Felt252Wrapper {
+    fn decode<I: Input>(input: &mut I) -> Result<Self, Error> {
+        let mut buf: [u8; 32] = [0; 32];
+        input.read(&mut buf)?;
+
+        match Felt252Wrapper::try_from(&buf) {
+            Ok(felt) => Ok(felt),
+            Err(e) => Err(Error::from("Can't get FieldElement from input buffer.").chain(hex::encode(buf)).chain(e)),
+        }
+    }
+}
+
+/// SCALE trait.
+impl TypeInfo for Felt252Wrapper {
+    type Identity = Self;
+
+    // The type info is saying that the field element must be seen as an
+    // array of bytes.
+    fn type_info() -> Type {
+        Type::builder()
+            .path(Path::new("Felt252Wrapper", module_path!()))
+            .composite(Fields::unnamed().field(|f| f.ty::<[u8]>().type_name("FieldElement")))
+    }
+}
+
+#[derive(Debug, PartialEq)]
+/// Error related to Felt252Wrapper.
+pub enum Felt252WrapperError {
+    /// Conversion from byte array has failed.
+    FromArrayError,
+    /// Provided byte array has incorrect lengths.
+    InvalidLength,
+    /// Invalid character in hex string.
+    InvalidCharacter,
+    /// Value is too large for FieldElement (felt252).
+    OutOfRange,
+}
+
+#[cfg(feature = "std")]
+impl std::error::Error for Felt252WrapperError {}
+
+use alloc::borrow::Cow;
+
+impl From<Felt252WrapperError> for Cow<'static, str> {
+    fn from(err: Felt252WrapperError) -> Self {
+        match err {
+            Felt252WrapperError::FromArrayError => Cow::Borrowed("input array invalid"),
+            Felt252WrapperError::InvalidCharacter => Cow::Borrowed("invalid character"),
+            Felt252WrapperError::OutOfRange => Cow::Borrowed("number out of range"),
+            Felt252WrapperError::InvalidLength => Cow::Borrowed("invalid length"),
+        }
+    }
+}
+
+#[cfg(feature = "std")]
+impl From<Felt252WrapperError> for std::string::String {
```

Comment:
I think this could be implemented in `no_std` too.
It just requires the feature using `alloc::string::String` rather than the `std` path.
If you really don't wish to make it available by default, put it behind `#[cfg(any(feature = "std", feature = "alloc"))]`

Reply:
It was more ignorance of my side. I read a bit about `no_std` and `alloc`, your comment is definitely the way to go. Thanks again @tdelabro.